projects/exui-common-lib/src/lib/components/find-service/find-service.component.ts

```diff
@@ -1,30 +1,97 @@
-import { Component, Input } from '@angular/core';
-import { FormBuilder, FormGroup } from '@angular/forms';
-import { FilterFieldConfig } from '../../models';
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { FormArray, FormControl, FormGroup } from '@angular/forms';
+import { FilterConfigOption, FilterFieldConfig } from '../../models';
 
 @Component({
   selector: 'xuilib-find-service',
   templateUrl: './find-service.component.html',
   styleUrls: ['./find-service.component.scss']
 })
-export class FindServiceComponent  {
-    @Input() public field: FilterFieldConfig;
-    @Input() public fields: FilterFieldConfig[];
-    @Input() public serviceTitle = 'Search for a service by name';
-    @Input() public form: FormGroup;
-    @Input() public services: any;
-    @Input() public selectedServices: any;
-    @Input() public disabled: any;
-    @Input() public enableAddServiceButton: boolean = true;
-    @Input() public disableInputField = false;
-
-    constructor(private readonly fb: FormBuilder) {
-      this.form = this.fb.group({
-        searchTerm: ['']
-      });
+export class FindServiceComponent implements OnInit {
+  @Input() public field: FilterFieldConfig;
+  @Input() public fields: FilterFieldConfig[];
+  @Input() public serviceTitle = 'Choose a service';
+  @Input() public form: FormGroup;
+  @Input() public services: FilterConfigOption[] = [];
+  @Input() public selectedServices: FilterConfigOption[] = [];
+  @Input() public enableAddServiceButton: boolean = true;
+  @Input() public disabled: boolean = false;
+
+  @Output() public serviceFieldChanged = new EventEmitter<void>();
+  @Output() public onError = new EventEmitter<void>();
```

Comment:
You don't need @Output() error. this is managed in the FormGroup/FilterService

Reply:
As discussed, I have removed it.